api/src/main/java/org/asynchttpclient/Request.java

```diff
@@ -112,6 +113,13 @@
     InputStream getStreamData();
 
     /**
+     * Return the current request's body as a Reactive Stream Publisher.
+     *
+     * @return a Reactive Stream Publisher representation of the current request's body.
```

Comment:
> This will return null if no publisher was set, right?

I'm not sure what is the contract of `Request`. Though, by looking at `RequestBuilder`, it seems `null` is an acceptable value to return in all the getters in `Request`.

>  we definitely don't want to expose any arbitrarily set body as a publisher as implementing that will require either depending on a reactive streams implementation from the async-http-client API, or implementing reactive streams ourselves.

Here I'm not following. The AHC client already depends on reactive-stream, and the `setBody(Publisher<T>)` would be only a conveniency method so that you can write `client.preparePut(getTargetUrl()).setBody(LARGE_IMAGE_PUBLISHER)` instead of `client.preparePut(getTargetUrl()).setBody(new ReactiveStreamBodyGenerator(LARGE_IMAGE_PUBLISHER))`. That said, I can see the argument of not polluting the `Request` interface with yet another method.

@slandelle WDYT?